kroxylicious-integration-test-support/src/main/java/io/kroxylicious/test/MirrorImageNameSubstitutor.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.test;
+
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.ImageNameSubstitutor;
+
+public class MirrorImageNameSubstitutor extends ImageNameSubstitutor {
+    private static final Set<String> DOMAIN_WHITELIST = Set.of("quay.io", "ghcr.io", "gcr.io");
+    private static final String MIRROR_REPOSITORY = "mirror.gcr.io";
+    private static final Logger LOGGER = LoggerFactory.getLogger(MirrorImageNameSubstitutor.class);
+
+    @Override
+    public DockerImageName apply(DockerImageName dockerImageName) {
+        if (DOMAIN_WHITELIST.stream().anyMatch(domain -> dockerImageName.getRegistry().endsWith(domain))) {
+            return dockerImageName;
+        }
+        else {
+            DockerImageName substituted = dockerImageName.withRegistry(MIRROR_REPOSITORY);
+            LOGGER.warn("replacing image repository for {} with mirror registry: {}", dockerImageName, substituted);
+            return substituted;
+        }
```

Comment:
This effectively assumes that everything not in the whitelist is included in the mirror (and by extension is effectively from dockerhub). 

I do wonder if we would be better served by a map of original to mirror, where the current allow list could be supported by an identity default. e.g. the default is not to replace registries.